src/Language/PureScript/Linter/Exhaustive.hs

```diff
@@ -108,31 +108,35 @@ genericMerge f bsl@((s, b):bs) bsr@((s', b'):bs')
 
 -- |
 -- Find the uncovered set between two binders:
--- the first binder is the case we are trying to cover the second one is the matching binder
+-- the first binder is the case we are trying to cover, the second one is the matching binder
 --
-missingCasesSingle :: Environment -> ModuleName -> Binder -> Binder -> [Binder]
-missingCasesSingle _ _ _ NullBinder = []
-missingCasesSingle _ _ _ (VarBinder _) = []
+missingCasesSingle :: Environment -> ModuleName -> Binder -> Binder -> ([Binder], Maybe Bool)
+-- Maybe Bool: Nothing = DontKnow, Just False = Redundant (covers nothing)
+--             Just True = NotRedundant (covers something)
+missingCasesSingle _ _ _ NullBinder = ([], Just True)
+missingCasesSingle _ _ _ (VarBinder _) = ([], Just True)
 missingCasesSingle env mn (VarBinder _) b = missingCasesSingle env mn NullBinder b
 missingCasesSingle env mn br (NamedBinder _ bl) = missingCasesSingle env mn br bl
 missingCasesSingle env mn NullBinder cb@(ConstructorBinder con _) =
-  concatMap (\cp -> missingCasesSingle env mn cp cb) allPatterns
+  (concatMap (\cp -> fst $ missingCasesSingle env mn cp cb) allPatterns, Just True)
   where
   allPatterns = map (\(p, t) -> ConstructorBinder (qualifyName p mn con) (initialize $ length t))
                   $ getConstructors env mn con
 missingCasesSingle env mn cb@(ConstructorBinder con bs) (ConstructorBinder con' bs')
-  | con == con' = map (ConstructorBinder con) (missingCasesMultiple env mn bs bs')
-  | otherwise = [cb]
-missingCasesSingle _ _ NullBinder (ArrayBinder bs)
-  | null bs = [] 
-  | otherwise = []
+  | con == con' = let (bs'', pr) = missingCasesMultiple env mn bs bs' in (map (ConstructorBinder con) bs'', pr)
+  | otherwise = ([cb], Just False)
+missingCasesSingle _ _ NullBinder (ArrayBinder _) = ([], Nothing)
```

Comment:
This case doesn't seem right. It would determine `f [] = 0` to be exhaustive, wouldn't it?